hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -186,10 +192,31 @@ default boolean closeIsCooperative() {
      *              Note that it might not be the actual error that caused the job
      *              to fail - it can be several other exceptions. We only guarantee
      *              that it's non-null if the job didn't complete successfully.
+     * @see #isReusable()
      */
     default void close(@Nullable Throwable error) throws Exception {
     }
 
+    /**
+     * Returns {@code true} if this instance can be used in different vertices
+     * or in different job executions. In that case, {@link #init}, {@link #get}
+     * and {@link #close} methods must be idempotent and thread-safe.
```

Comment:
"idempotent" may be misleading, eg. `get` can always return new function, eg. for RandomMemberPms it behaves differently each time, so it is not idempotent but is reusable.